com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -217,18 +207,19 @@ internal static void ChangeOwnership(NetworkObject networkObject, ulong clientId
                 throw new SpawnStateException("Object is not spawned");
             }
 
-            if (NetworkManager.Singleton.ConnectedClients.ContainsKey(networkObject.OwnerClientId))
+            if (NetworkManager.Singleton.ConnectedClients.TryGetValue(networkObject.OwnerClientId, out NetworkClient networkClient))
             {
-                for (int i = NetworkManager.Singleton.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.Count - 1; i >= 0; i--)
+                for (int i = networkClient.OwnedObjects.Count - 1; i >= 0; i--)
                 {
-                    if (NetworkManager.Singleton.ConnectedClients[networkObject.OwnerClientId].OwnedObjects[i] == networkObject)
+                    if (networkClient.OwnedObjects[i] == networkObject)
                     {
-                        NetworkManager.Singleton.ConnectedClients[networkObject.OwnerClientId].OwnedObjects.RemoveAt(i);
+                        networkClient.OwnedObjects.RemoveAt(i);
                     }
                 }
+
+                networkClient.OwnedObjects.Add(networkObject);
```

Comment:
I moved this line into the TryGetValue. This is technically a bug fixe because before it would throw a KeyNotFoundException which makes no sense because it was **immediately** after the ContainsKey if branch for checking whether the key existed.